.github/workflows/callable-e2e-tests-headless.yml

```diff
@@ -0,0 +1,72 @@
+name: 'E2E Tests - Headless'
+
+on:
+  workflow_call:
+    inputs:
+      test_name:
+        required: true
+        type: string
+      category:
+        required: true
+        type: string
+      spec:
+        required: true
+        type: string
+      timeout_minutes:
+        required: true
+        type: number
+      retry_count:
+        required: true
+        type: number
+
+jobs:
+  e2e-test:
+    name: 'E2E-Headless: ${{ inputs.test_name }}'
+    runs-on: ubuntu-latest
+    timeout-minutes: ${{ inputs.timeout_minutes }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@24cb9080177205b6e8c946b17badbe402adc938f # v3.4.0 https://github.com/actions/checkout/commit/24cb9080177205b6e8c946b17badbe402adc938f
+        with:
+          # Minimal depth 2 so we can checkout the commit before possible merge commit.
+          fetch-depth: 2
+          path: aws-amplify
+
+      - name: Setup node and build the repository
+        uses: ./aws-amplify/.github/actions/node-and-build
+
+      - name: Setup samples staging repository
+        uses: ./aws-amplify/.github/actions/setup-samples-staging
+        with:
+          GH_PAT: ${{ secrets.GH_PAT }}
+
+      - name: 'Change directory into build and run link'
+        run: |
+          yarn link-all
+        working-directory: aws-amplify
+
+      - name: 'Link packages into samples staging root'
+        run: |
+          yarn link amazon-cognito-identity-js @aws-amplify/core @aws-amplify/cache @aws-amplify/auth
+        working-directory: amplify-js-samples-staging
+
+      - name: Run cypress tests for ${{ inputs.test_name }}
+        run: |
+          ../aws-amplify/.circleci/retry-yarn-script.sh -s \
```

Comment:
Two things.
1. The `aws-amplify` part of this path shouldn't work anymore, right? Should be `amplify-js` now.
2. As a separate change, can we copy this script and anything else from .circleci into scripts (or somewhere), retaining the `chmod` perm level, but moved so this directory can be deleted?

Reply:
sure it'll be good to move `retry-yarn-script.sh` to a different dir. 
let's discuss this